### ISSUE TITLE
Missing derivations in VLog

### ISSUE THREAD

**Issue by mkroetzsch** [karmaresearch/vlog#58]
On a specific input, I am missing a derivation from VLog. The input KB in Rulewerk syntax is:
```
TRIPLE(i1, first, element1) .
TRIPLE(i1, rest, i2) .
TRIPLE(i2, first, element2) .
TRIPLE(i2, rest, nil) .
TRIPLE(c, intersectionOf, i1) .

element(?L, ?E) :- TRIPLE(?L, first, ?E) .
next(?L1, ?L2) :- TRIPLE(?L1, rest, ?L2), element(?L2, ?E) .
List(?Y), hasList(?X, ?Y) :- TRIPLE(?X, intersectionOf, ?Y) .
List(?Z) :- List(?Y), next(?Y, ?Z) .
```
This should derive List(i1) and List(i2), but only List(i1) is derived. If I change the order of rules or facts, or if I remove the "hasList" head atom, it works correctly. I believe that this is a VLog bug, since the query results and the query result count reported by VLog agree, so I do not have any indication that Rulewerk could somehow cause this. The full list of inferred facts, as queries through Rulewerk, is:
```
hasList(c, i1) .
next(i1, i2) .
TRIPLE(i1, first, element1) .
TRIPLE(i1, rest, i2) .
TRIPLE(i2, first, element2) .
TRIPLE(i2, rest, nil) .
TRIPLE(c, intersectionOf, i1) .
element(i1, element1) .
element(i2, element2) .
List(i1) .
```

**Comment by mkroetzsch** [comment]
The bug can be reproduced in VLog on the command line. Rules:
```
mytriple(L,T,E) :- TRIPLE(L,T,E)
element(L,E) :- TRIPLE(L,first,E)
next(L1,L2) :- TRIPLE(L1,rest,L2),element(L2,E)
last(L1) :- TRIPLE(L1,rest,nil)
Intersection(Y) :- TRIPLE(X,intersectionOf,Y)
hasList(X,Y)   :- TRIPLE(X,intersectionOf,Y)
Union(Y)        :- TRIPLE(X,unionOf,Y)
hasList(X,Y)   :- TRIPLE(X,unionOf,Y)
Intersection(Z) :- Intersection(Y),next(Y,Z)
Union(Z) :- Union(Y),next(Y,Z)
hasList(X,Z) :- hasList(X,Y),next(Y,Z)
hasIntersectionElement(X,E) :- hasList(X,Y),Intersection(Y),element(Y,E)
hasUnionElement(X,E) :- hasList(X,Y),Union(Y),element(Y,E)
```
EDB configuration edb.conf:
```
EDB0_predname=TRIPLE
EDB0_type=INMEMORY
EDB0_param0=.
EDB0_param1=triples
```
Data file triples.csv contains:
```
B-fa5724d6-ca7b-30d2-8444-297a0b98ebc3,intersectionOf,B-360d4bb5-c9d3-3c12-8f6d-59e7651598e2
B-360d4bb5-c9d3-3c12-8f6d-59e7651598e2,first,Woman
B-360d4bb5-c9d3-3c12-8f6d-59e7651598e2,rest,B-1a790bdd-6a06-3bfc-8567-7703fdad45f6
B-1a790bdd-6a06-3bfc-8567-7703fdad45f6,first,Parent
B-1a790bdd-6a06-3bfc-8567-7703fdad45f6,rest,nil
```

**Comment by CerielJacobs** [comment]
Fixed by fixing issue #59 .
